pyiceberg/avro/resolver.py

```diff
@@ -233,7 +255,107 @@ def skip(self, decoder: BinaryDecoder) -> None:
         pass
 
 
-class SchemaResolver(PrimitiveWithPartnerVisitor[IcebergType, Reader]):
+class WriteSchemaResolver(PrimitiveWithPartnerVisitor[IcebergType, Writer]):
+    def schema(self, schema: Schema, expected_schema: Optional[IcebergType], result: Writer) -> Writer:
+        return result
+
+    def struct(self, struct: StructType, provided_struct: Optional[IcebergType], field_writers: List[Writer]) -> Writer:
```

Comment:
I think it would be easier to follow the logic here if you used `file_` as a prefix for all of the outgoing types. I keep having to remember which one is the file schema.

Reply:
Yes, good one. I've updated it to `data_schema` since it is more likely that it comes from the memory.